src/functions.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+function displayBooks($books) {
+    $htmlOutput = ''; 
+    foreach ($books as $book) {
+        if (isset($book['authorFirstName']) 
+        && isset($book['authorSecondName'])
+        && isset($book['bookTitle'])
+        && isset($book['textType'])
+        && isset($book['genre'])
+        && isset($book['yearPublished'])
+        && isset($book['authorInspiration'])
+        && isset($book['authorFirstName'])
+        && isset($book['keyQuote'])
+        && isset($book['languageWrittenIn'])
+        && isset($book['link'])
+        && isset($book['extra']) 
+        && $book['authorFirstName']
+        && $book['authorSecondName']
+        && $book['bookTitle']
+        && $book['textType']
+        && $book['genre']
+        && $book['yearPublished']
+        && $book['authorInspiration']
+        && $book['authorFirstName']
+        && $book['keyQuote']
+        && $book['languageWrittenIn']
+        && $book['link']
+        && $book['extra']) {
```

Comment:
this validation plus the exception logic should be in a separate function called something like `validateBook`, which either returns true or throws an exception. then that function needs to be unit tested as well